drivers: i2c: i2c_mcux: unused variables cause daily build failures

**Describe the bug**
Twister fails during the daily build due to unused variables (warnings promoted to errors).

**To Reproduce**
```
twister -i -p ip_k66f -s samples/subsys/display/grove_display/sample.display.grove_display
```

**Expected behavior**
Daily build should not fail.

**Impact**
Annoyance, because the build succeeds with
```
west build -p auto -b ip_k66f samples/subsys/display/grove_display
```

**Logs and console output**
```
/__w/zephyr/zephyr/drivers/i2c/i2c_mcux.c:216:36: error: 'i2c_mcux_driver_api' defined but not used [-Werror=unused-const-variable=]
  216 | static const struct i2c_driver_api i2c_mcux_driver_api = {
      |                                    ^~~~~~~~~~~~~~~~~~~
/__w/zephyr/zephyr/drivers/i2c/i2c_mcux.c:186:12: error: 'i2c_mcux_init' defined but not used [-Werror=unused-function]
  186 | static int i2c_mcux_init(const struct device *dev)
      |            ^~~~~~~~~~~~~
/__w/zephyr/zephyr/drivers/i2c/i2c_mcux.c:178:13: error: 'i2c_mcux_isr' defined but not used [-Werror=unused-function]
  178 | static void i2c_mcux_isr(const struct device *dev)
      |             ^~~~~~~~~~~~
cc1: all warnings being treated as errors
```

**Environment (please complete the following information):**
 - OS: CI
 - Toolchain: Zephyr SDK 0.13.2
 - Commit SHA: 7f5eb921dc8bb1b7014886986bbffdf111b68b15

**Additional context**
Daily Build failure:
https://github.com/zephyrproject-rtos/zephyr/runs/4676613761?check_suite_focus=true

--- a/samples/subsys/display/grove_display/sample.yaml
+++ b/samples/subsys/display/grove_display/sample.yaml
@@ -5,3 +5,4 @@ tests:
     tags: drivers
     harness: grove
     depends_on: i2c
+    platform_exclude: ip_k66f

